app/core/downloader/provider_resolution.py

```
@@ -0,0 +1,168 @@
+def _try_get_direct(ep: Episode, provider_name: str, language: str) -> Optional[str]:
+    language = normalize_language(language)
+    logger.info("Trying provider '%s' for language '%s'", provider_name, language)
+    try:
+        url = ep.get_direct_link(provider_name, language)  # Lib-API
+        if url:
+            logger.success(
+                "Found direct URL from provider '%s': %s", provider_name, url
+            )
+            return url
+        logger.warning("Provider '%s' returned no URL.", provider_name)
+    except Exception as exc:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)